### PR TITLE
[10.0] Add account_mass_reconcile_partner

### PR DESCRIPTION
**Pull request by grindtildeath**
This module extends the functionality of account_mass_reconcile and adds a new
reconciliation method Advanced. Partner.
With this method credit lines are matched with debit lines in a chronological
order if the partner is the same.

### CODE REVIEW

**Review by vrenaville — APPROVED** [review]

File: account_mass_reconcile_partner/models/mass_reconcile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+class AccountMassReconcileMethod(models.Model):
+    _inherit = 'account.mass.reconcile.method'
+
+    @api.model
+    def _get_all_rec_method(self):
+        res = super(AccountMassReconcileMethod, self)._get_all_rec_method()
+        res.append((
+            'mass.reconcile.advanced.partner',
+            'Advanced. Partner'
```
Comment (simahawk) [inline account_mass_reconcile_partner/models/mass_reconcile.py:15]: "Advanced. Partner" correct w/ that dot?

File: account_mass_reconcile_partner/tests/test_reconcile_partner.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import time
+
+from odoo.tests import SavepointCase
+
+
+class TestAccountReconcilePartnerChrono(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestAccountReconcilePartnerChrono, cls).setUpClass()
```
Comment (simahawk) [inline account_mass_reconcile_partner/tests/test_reconcile_partner.py]: pls disable tracking suite-wise if you don't need it

**Review by sebalix — APPROVED** [review]

**Review by simahawk — APPROVED** [review]
LG, non blocking things to address if you can.

**Review by i-vyshnevska — APPROVED** [review]

File: account_mass_reconcile_partner/models/mass_reconcile.py (hunk 1 above)
Comment (grindtildeath) [reply account_mass_reconcile_partner/models/mass_reconcile.py:15]: Yes, it's to stay consistent with the existing : https://github.com/OCA/account-reconcile/blob/11.0/account_mass_reconcile/models/mass_reconcile.py#L81

File: account_mass_reconcile_partner/models/__init__.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+from . import advanced_reconciliation
```
Comment (umiphos) [inline account_mass_reconcile_partner/models/__init__.py:1]: Can you add the License?

example:
https://github.com/OCA/maintainer-tools/blob/master/template/module/models/__init__.py

Comment (sebalix) [reply account_mass_reconcile_partner/models/__init__.py:1]: It's not required for __init__.py where there is only import statements.

Comment (guewen) [reply account_mass_reconcile_partner/models/__init__.py:1]: I don't feel that we need a license on import statements.

Comment (simahawk) [reply account_mass_reconcile_partner/models/__init__.py:1]: same here :)

**Comment by fclementic2c** [comment]
Functional test ok :+1:

**Review by fclementic2c — APPROVED** [review]

**Pull request merged**